desktop/core/src/desktop/api_public.py

```diff
@@ -122,6 +122,20 @@ def get_logs(request):
   return notebook_api.get_logs(django_request)
 
 
+@api_view(["POST"])
+def get_sample_data(request, server=None, database=None, table=None, column=None):
+  django_request = get_django_request(request)
+
+  data = {
+    'snippet': '{"type":"mysql","result":{}}',
+  }
+
+  django_request.POST = QueryDict(mutable=True)
+  django_request.POST.update(data)
```

Comment:
@romainr will it be better to make change [patch_operation_id_request](https://github.com/cloudera/hue/blob/3bf8ce3fa0c3a1987b13e82d565ad8e1d37d9cb8/desktop/core/src/desktop/api_public.py#L169) accepting dict like `{ snippet: ..., operation id: ...}` as params? The if check can be in view itself

Reply:
Hum, change it to check for each top attribute? 
```
data = {}
if not snippet
if not operationId
```

Reply:
Hum, change it to check for each top attribute? (in a separate commit, and update /sample to reuse then too)
```
data = {}
if not snippet
if not operationId
```

Reply:
PR #2301  sent!